spec/models/course/assessment_spec.rb

```diff
@@ -57,6 +57,34 @@
           it { is_expected.to be_valid }
         end
       end
+
+      context 'when the assessment is set to be autograded' do
+        subject { assessment }
+
+        context 'when the assessment has a non-autograded question' do
+          let!(:non_autograded_question) do
+            create(:course_assessment_question, assessment: assessment)
+          end
+
+          it 'is not valid' do
+            assessment.autograded = true
```

Comment:
I think it's fine then. Or you can build the question and skip validation while saving.